roles/lib_utils/library/docker_creds.py

```diff
@@ -135,7 +135,7 @@ def update_config(docker_config, registry, username, password):
         docker_config['auths'][registry] = {}
 
     # base64 encode our username:password string
-    encoded_data = base64.b64encode('{}:{}'.format(username, password))
```

Comment:
I think this is currently working as-is.  What is the extra .encode() for?

Reply:
NVM, I see this is another python3 gotcha.

Reply:
it fails on Python3 for me:

```
$ python3
Python 3.6.3 (default, Oct  9 2017, 12:07:10) 
[GCC 7.2.1 20170915 (Red Hat 7.2.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import base64
>>> user = "user"
>>> pw = "pass"
>>> base64.b64encode('{}:{}'.format(user, pw))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.6/base64.py", line 58, in b64encode
    encoded = binascii.b2a_base64(s, newline=False)
TypeError: a bytes-like object is required, not 'str'
```